webapp/channels/src/components/post_view/commented_on/commented_on.tsx

```diff
@@ -40,42 +41,39 @@ export default class CommentedOn extends PureComponent<Props> {
         return message;
     };
 
-    render() {
-        const message = this.makeCommentedOnMessage();
-        const parentPostUser = this.props.parentPostUser;
-        const parentPostUserId = (parentPostUser && parentPostUser.id) || '';
+    const message = makeCommentedOnMessage();
+    const parentPostUserId = parentPostUser?.id || '';
+
+    const parentUserProfile = (
+        <UserProfile
+            user={parentPostUser}
+            userId={parentPostUserId}
+            displayName={parentPostUser?.username}
+            hasMention={true}
+            disablePopover={false}
+        />
```

Comment:
In order to properly overwrite name and icon you should pass here the `overwriteName` and `overwriteIcon` props. Not sure if any other extra prop is needed to properly work.

Also, maybe I am misreading something, but looks like there is also a bug on `user_profile/user_profile.tsx` line 85-89, where if there is a display name it will use the username, and that the overwriteName is not checking the `enablePostUsernameOverride` option to add it.

Reply:
I don't think this `displayName` actually passed here makes it to the component because there's one added by `UserProfile`'s `mapStateToProps` based on the `userId`. `overwriteName` and `overwriteIcon` are used for posts made by integrations, so they're not needed in this case.

There is however a separate bug here where this message doesn't show the overwritten name/picture for those integration posts, but I've filed that separately (https://mattermost.atlassian.net/browse/MM-54190)

Reply:
https://github.com/mattermost/mattermost/pull/24322#discussion_r1300599844